pkg/services/authz/zanzana/server/server_mutate_roles.go

```diff
@@ -0,0 +1,120 @@
+	ctx, span := s.tracer.Start(ctx, "server.mutateRoles")
+	defer span.End()
+
+	writeTuples := make([]*openfgav1.TupleKey, 0)
+	deleteTuples := make([]*openfgav1.TupleKeyWithoutCondition, 0)
+
+	for _, operation := range operations {
+		switch op := operation.Operation.(type) {
+		case *authzextv1.MutateOperation_CreateRole:
+			tuples, err := getRoleTuples(op.CreateRole.RoleName, op.CreateRole.Permissions)
+			if err != nil {
+				return err
+			}
+			writeTuples = append(writeTuples, tuples...)
+		case *authzextv1.MutateOperation_DeleteRole:
+			tuples, err := getRoleTuples(op.DeleteRole.RoleName, op.DeleteRole.Permissions)
+			if err != nil {
+				return err
+			}
+			deletes := make([]*openfgav1.TupleKeyWithoutCondition, 0, len(tuples))
+			for _, tuple := range tuples {
+				deletes = append(deletes, common.ToOpenFGADeleteTupleKey(tuple))
+			}
+			deleteTuples = append(deleteTuples, deletes...)
+		default:
+			s.logger.Debug("unsupported mutate operation", "operation", op)
+		}
+	}
+
+	writeReq := &openfgav1.WriteRequest{
+		StoreId:              store.ID,
+		AuthorizationModelId: store.ModelID,
+	}
+	if len(writeTuples) > 0 {
+		writeReq.Writes = &openfgav1.WriteRequestWrites{
+			TupleKeys:   writeTuples,
+			OnDuplicate: "ignore",
+		}
+	}
+	if len(deleteTuples) > 0 {
+		writeReq.Deletes = &openfgav1.WriteRequestDeletes{
+			TupleKeys: deleteTuples,
+			OnMissing: "ignore",
+		}
+	}
+
+	_, err := s.openfga.Write(ctx, writeReq)
+	if err != nil {
+		s.logger.Error("failed to write resource role binding tuples", "error", err)
+		return err
+	}
+
+	return nil
+}
+
+func getRoleTuples(roleName string, rolePermissions []*authzextv1.RolePermission) ([]*openfgav1.TupleKey, error) {
+	permissions := make([]iamv0.RolespecPermission, len(rolePermissions))
+	for i, p := range rolePermissions {
+		permissions[i] = iamv0.RolespecPermission{
+			Action: p.Action,
+			Scope:  p.Scope,
+		}
+	}
+	return convertRolePermissionsToTuples(roleName, permissions)
+}
+
+// convertRolePermissionsToTuples converts role permissions (action/scope) to v1 TupleKey format
+// using the shared zanzana.ConvertRolePermissionsToTuples utility and common.ToAuthzExtTupleKeys
+func convertRolePermissionsToTuples(roleUID string, permissions []iamv0.RolespecPermission) ([]*openfgav1.TupleKey, error) {
+	// Convert IAM permissions to zanzana.RolePermission format
+	rolePerms := make([]zanzana.RolePermission, 0, len(permissions))
+	for _, perm := range permissions {
+		// Split the scope to get kind, attribute, identifier
+		kind, _, identifier := splitScope(perm.Scope)
+		rolePerms = append(rolePerms, zanzana.RolePermission{
+			Action:     perm.Action,
+			Kind:       kind,
+			Identifier: identifier,
+		})
+	}
+
+	// Translate to Zanzana tuples
+	tuples, err := zanzana.ConvertRolePermissionsToTuples(roleUID, rolePerms)
+	if err != nil {
+		return nil, err
+	}
+
+	return tuples, nil
+}
```

Comment:
What's the benefit of the intermediate `iamv0` conversion?
```suggestion
// convertRolePermissionsToTuples converts role permissions (action/scope) to v1 TupleKey format
// using the shared zanzana.ConvertRolePermissionsToTuples utility and common.ToAuthzExtTupleKeys
func convertRolePermissionsToTuples(roleUID string, permissions []*authzextv1.RolePermission) ([]*openfgav1.TupleKey, error) {
	// Convert to zanzana.RolePermission
	rolePerms := make([]zanzana.RolePermission, 0, len(permissions))
	for _, perm := range permissions {
		// Split the scope to get kind, attribute, identifier
		kind, _, identifier := splitScope(perm.Scope)
		rolePerms = append(rolePerms, zanzana.RolePermission{
			Action:     perm.Action,
			Kind:       kind,
			Identifier: identifier,
		})
	}

	// Translate to Zanzana tuples
	tuples, err := zanzana.ConvertRolePermissionsToTuples(roleUID, rolePerms)
	if err != nil {
		return nil, err
	}

	return tuples, nil
}
```